### PR TITLE
[BreakingChange] Update core to use tokens. Update and add new components

### LINKED ISSUE #124: Allow access to Button components underlying button node by supporting ref
**Is your feature request related to a problem? Please describe.**
React offers the `ref` mechanism for communicating with the underlying DOM nodes. [The React documents recommend this mechanism for managing `focus`](https://reactjs.org/docs/refs-and-the-dom.html). In Varaamo, I was trying to alter the natural tab order on the page in order to skip content that was not accessible with keyboard. In my case, the next component after the skipped content was a button which I wanted to focus like so: `ref.current.focus()`.

**Describe the solution you'd like**
[React.forwardRef](https://reactjs.org/docs/react-api.html#reactforwardref) allows to easily manage refs. After this change I would be able to pass down a `ref` in the `ref` prop of the `Button` tag.

**Describe alternatives you've considered**
`refs` could be enabled for other components as well. Another important candidate would be `TextInput`.

### PR DESCRIPTION
**Pull request by niglu1**
## Description

This is a pretty big update and it contains some breaking changes to the core and react packages.

### Highlights
- hds-core and hds-react now use tokens from hds-design-tokens
- Updated Button and TextInput components to new design
- New components
  - TextArea
  - RadioButton
  - Checkbox
- You can now pass props to the native element for Button, TextInput  (and the new components)
- `ref` is now forwarded for Button and TextInput (and the new components)

## Changes

**Design Tokens:**
- Added `lineheight` and `fontsize` typography tokens
- Added new `light-5` tokens for bus and coat-blue brand colors
- Added new `-dark` tokens for some UI helper colors
- Updated tokens for Button
- Updated tokens for TextInput
- Added tokens for RadioButton
- Added tokens for Checkbox

**Core:**
- Added new `base.css` file that uses design tokens.
- Removed `helsinki.css` and other files under `variables` as they are replaced by `base.css`
- Rewrote styles for Button
- Rewrote styles for TextInput and Textarea
- Added styles for RadioButton
- Added styles for Checkbox
- Updated Notification to use tokens

**React:**
- Refactored Button component
- Refactored TextInput component
- Added TextArea component
- Added RadioButton component
- Added Checkbox component
- Updated the rest of the components to use tokens

**Site:**
- Added pages for the new components
- Removed all old `--hds` variables and replaced them with tokens
- Added typography tokens

## Breaking changes

**Core:**
- `helsinki.css` and all the theming variables (`--hds-`) have been removed. The new `base.css` file replaces `helsinki.css`. It contains the base styles and imports needed.
- [Button] - Removed tertiary buttons
- [TextInput] - Removed alternative style

**React:**
- [Button] - `color` prop was renamed to `variant`
- [TextInput] - `alternative` prop was removed
- [TextInput] - `invalidText` prop was removed. Use `helperText` instead.

## Related issues

Closes #124, #130, #131

## How Has This Been Tested?

On storybook. The updated components can be tested here https://niglu1.github.io/?path=/story/components-button--primary

### CODE REVIEW

File: packages/core/src/components/notification/notification.css [hunk 1]
```diff
@@ -1,9 +1,9 @@
 .hds-notification {
   margin: 0.75em 0 1.25em;
   padding: 1.875em 1.875em 1.875em;
-  border: 0 solid var(--hds-theme-color-primary);
+  border: 0 solid var(--color-coat-of-arms-blue);
   border-left-width: 0.625em;
-  background-color: var(--hds-theme-color-primary-extra-light);
+  background-color: var(--color-coat-of-arms-blue-light-20);
```
Comment (lehtoinen) [inline packages/core/src/components/notification/notification.css:6]: What if I want to use the `Notification` but with other colors than Coat of Arms? I guess here I would overwrite the `--color-coat-of-arms-blue` var, but that seems a bit hacky.

Should these be more generic vars (something like `--notification-border-color` and `--notification-background-color` or even `--color-primary` & `--color-primary-light-20`) with the `coat-of-arms`-colors as defaults?

Comment (niglu1) [reply packages/core/src/components/notification/notification.css]: Valid point. I updated all the components that were still using `--hds` or hard coded color variables to support defining colors for them at the top level. So e.g. with `Notification` you can now easily change any color by overriding any of the variables

```css
.hds-notification {
  --background-color-default: var(--color-coat-of-arms-blue-light-20);
  --background-color-success: var(--color-success-light);
  --background-color-warning: var(--color-alert-light);
  --background-color-error: var(--color-error-light);

  --border-default: var(--color-coat-of-arms-blue);
  --border-success: var(--color-success);
  --border-warning: var(--color-alert);
  --border-error: var(--color-error);

  --close-button-default: var(--color-coat-of-arms-blue-dark-50);
  --close-button-success: var(--color-success-dark);
  --close-button-warning: var(--color-alert-dark);
  --close-button-error: var(--color-error-dark);
}
```

File: packages/core/src/components/button/button.stories.js [hunk 2]
```diff
@@ -3,18 +3,19 @@ import '../../icons/icon.css';
...
 
 /**
  * Primary
  */
 export const Primary = () => `
   <button type="button" class="hds-button hds-button--primary">
-    ${label}
+    ${getLabel()}
   </button>
 `;
```
[the remark this answers is not in the document]
Comment (niglu1) [reply packages/core/src/components/button/button.stories.js]: I removed all the comments from the story-files.

File: packages/core/README.md [hunk 3]
```diff
@@ -24,29 +32,22 @@ hds-core/
...
+    └── base.css        # base styles
 ```
 
 ## Using the styles
 
-At the minimum you'll need two stylesheets to use the core components: `helsinki.css` (the base theme) and `components/all.css`. If you use any of the icons, you'll also need `icons/all.css`. Do note that the `all.css`-files bundle all the components and icons into single stylesheets and thus are large in filesize.
+At the minimum you'll need two stylesheets to use the core components: `base.css` (the base styles and CSS variables) and `components/all.css`. If you use any of the icons, you'll also need `icons/all.css`. Do note that the `all.css`-files bundle all the components and icons into single stylesheets and thus are large in filesize.
 
-Therefore, we do recommend that you use individual stylesheets where needed. For reference, see the files in `examples` folder. For example [buttons.html](examples/buttons.html).
+Therefore, we do recommend that you use individual stylesheets where needed.
```
[the remark this answers is not in the document]
Comment (niglu1) [reply packages/core/README.md]: 👍

File: packages/core/README.md [hunk 4]
```diff
@@ -24,29 +32,22 @@ hds-core/
...
-
-```
-$ yarn add hds-core
+    └── base.css        # base styles
 ```
 
 ## Using the styles
 
-At the minimum you'll need two stylesheets to use the core components: `helsinki.css` (the base theme) and `components/all.css`. If you use any of the icons, you'll also need `icons/all.css`. Do note that the `all.css`-files bundle all the components and icons into single stylesheets and thus are large in filesize.
+At the minimum you'll need two stylesheets to use the core components: `base.css` (the base styles and CSS variables) and `components/all.css`. If you use any of the icons, you'll also need `icons/all.css`. Do note that the `all.css`-files bundle all the components and icons into single stylesheets and thus are large in filesize.
```
[the remark this answers is not in the document]
Comment (niglu1) [reply packages/core/README.md]: 👍

File: packages/core/README.md [hunk 5]
```diff
@@ -1,9 +1,17 @@
...
 Base styles and individual HTML component styles for the Helsinki Design System.
 
 ## Getting started
 
+### Installing the package:
+
+```
+$ yarn add hds-core
+```
+
```
[the remark this answers is not in the document]
Comment (niglu1) [reply packages/core/README.md:14]: I added a Development section to the readme with a link to the `DEVELOPMENT.md` file.

File: packages/design-tokens/README.md [hunk 6]
```diff
@@ -4,36 +4,37 @@ Design tokens for the Helsinki Design System.
 
 ## Getting started
 
+### Installing the package:
+
+```
+$ yarn add hds-design-tokens
+```
+
```
[the remark this answers is not in the document]
Comment (niglu1) [reply packages/design-tokens/README.md:12]: I added a section about updating and creating new tokens.

File: packages/design-tokens/tokens/color/component/radio-button/border.json [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        "selected-hover": {
+          "value": "{color.bus-dark-50.value}",
+          "type": "color"
+        },
+        "selected-disabled": {
+          "value": "{color.black-20.value}",
+          "type": "color"
+        },
+        "unselected": {
+          "value": "#c4c4c4",
```
[the remark this answers is not in the document]
Comment (niglu1) [reply packages/design-tokens/tokens/color/component/radio-button/border.json]: I seems that the color was updated in the designs. I updated all tokens that used `#c4c4c4` to use the correct color (that already had a token).

File: packages/react/src/components/button/Button.tsx [hunk 8]
```diff
@@ -42,48 +48,55 @@ export type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
...
+    return (
+      <button
+        ref={ref}
+        disabled={disabled}
+        type="button"
+        className={classNames(
+          styles.button,
+          styles[variant],
+          styles[theme],
+          styles[size],
```
[the remark this answers is not in the document]
Comment (niglu1) [reply packages/react/src/components/button/Button.tsx]: Good catch! I prefixed the class names to ensure that there won't be any problems caused by the clashing names. I also checked that this was only affecting the `Button` component.

File: packages/core/src/components/checkbox/checkbox.css [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+  appearance: none;
+  -moz-appearance: none;
+  -webkit-appearance: none;
+  height: var(--size);
+  left: 0;
+  outline: none;
+  position: absolute;
+  top: 0;
+  width: var(--size);
+}
```
Comment (lehtoinen) [inline packages/core/src/components/checkbox/checkbox.css:71]: The cursor in not a pointer for the input:


```suggestion
.hds-checkbox .hds-checkbox__input {
  appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  height: var(--size);
  left: 0;
  outline: none;
  position: absolute;
  top: 0;
  width: var(--size);
  cursor: pointer;
}
```

...or then bump the z-index for label, if that's intended to handle the cursor state.

Comment (niglu1) [reply packages/core/src/components/checkbox/checkbox.css]: Whoopsie-daisy 😅

**Pull request merged**